### PR TITLE
Add Widnows CI jobs.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

Currently this just builds and packages the agent and then uploads the package as a workflow artifact for later inspection.

Additionally:

- Move all Windows-related packaging scripts into `packaging/windows` so it’s clear that they are _for Windows packaging_.
- Tidy up the scripts (basic dead code removal and syntax cleanup to make them consistent with the style of our other scripts).
- Infer the repository path from the path to the scripts instead of blindly assuming that the current working directory is the repository root. This makes the scripts significantly more resilient against issues with the environment.
- Separate the dependency installation code out as it’s own script so that it can be more clearly inferred what, exactly, is being done when it gets invoked.
- Ensure that MSYS2 is up to date when installing dependencies. This is required because it’s not actually possible to insist on an exact reproducible environment with MSYS2 due to the fact that they use a rolling release model combined with major limitations in their package manager. Note that _two_ passes may be needed for dependency installation depending on what needs updated, as MSYS2 will kill itself to ensure certain things get fully updated before continuing if specific packages need updated.
- Drop packages that are only needed for development from the list of build time dependencies. We do not need GDB or CCache in CI, so installing them is a massive waste of time (especially since `pacman` is so horribly inefficient).
- Use an absolute path for the `makensis` command so that we don’t rely on a customized environment for the scripts to run.
- Provide a proper fallback for the build directory, and also respect any build directory the script is asked to use.
- Always use the latest MSYS2 installer, instead of hard-coding the version. This is required for the same reasons outlined above about ensuring that things are updated as part of the dependency install process.
- Add new PowerShell scripts that can be used to invoke the build process from outside of MSYS2 (they’ll find the existing install of MSYS2 and use it). This significantly simplifies the CI job and also makes it much easier to locally test _exactly_ what CI is doing (which is extremely important if you
- Default to using a working configuration in the build script for the case of making a release build, and provide a way to pass extra options to CMake without modifying the script. This is important to keep the script clean and understandable, and to ensure that people can test what CI is doing locally without having to understand what CI is doing.

Because of the PowerShell scripts, one can now run a test build locally on any Windows install by opening a PowerShell session and running the following from the repository root:

```powershell
Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process
.\packaging\windows\install-dependencies.ps1
.\packaging\windows\build.ps1
.\packaging\windows\package.ps1
```

##### Test Plan

CI passes on this PR, and the produced installer actually works.

Manual testing will be needed to confirm that the installer is working consistently with the existing builds. It can be downloaded as the `windows-x86_64-installer` artifact on the Build workflow for this PR.

##### Additional Notes

This PR drops both the installation of the Mold linker and the CMake option that disables use of it. The net effect is that we don’t waste time on it at all in clean environments, but as a side effect anybody who was working with an environment set up with the scripts from _before_ this PR got merged will need to uninstall Mold to ensure it doesn’t get used.

This PR also changes the `ninja` invocations to be consistent with the changes that were made elsewhere in #18101. See that PR for the rationale behind this.

### CODE REVIEW

File: packaging/windows/build.ps1 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#Requires -Version 4.0
+
+$ErrorActionPreference = "Stop"
+
+. "$PSScriptRoot\functions.ps1"
+
+$msysbash = Get-MSYS2Bash "$msysprefix"
+$env:CHERE_INVOKING = 'yes'
+
+& $msysbash -l "$PSScriptRoot\compile-on-windows.sh"
```
Comment (stelfrag) [inline packaging/windows/build.ps1:12]: This probably needs to change to `compile-on-windows.sh package` to avoid a 1067 error on startup.

**Comment by thiagoftsm** [comment]
@Ferroin ,

Please, rebase your PR.

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts.

File: packaging/windows/build.ps1 (hunk 1 above)
Comment (Ferroin) [reply packaging/windows/build.ps1:12]: If we need to have the BUILD_FOR_PACKAGING option set for things to actually work, then that should be the default, not something that needs a explicit opt-in. This is the exact type of thing I was talking about when I said yesterday that we still needed a lot of work on the Windows build infrastructure to make it production quality.

Comment (Ferroin) [reply packaging/windows/build.ps1:12]: Or, looking at this alternatively, maybe the code that we’re controlling with this should be keyed off of the target platform, not a largely unrelated configuration option that was just intended to modify how the _build system_ behaves and not how the compiled binary behaves.

**Comment by thiagoftsm** [comment]
@Ferroin , our CI still is showing errors for Windows.

**Comment by thiagoftsm** [comment]
@Ferroin,

When I tried to compile locally using `msys`, I am having this error now:

```sh
+ '[' -d /home/thiago/Netdata/netdata/build-msys-MSYS ']'
./packaging/windows/compile-on-windows.sh: line 28: EXTRA_CMAKE_OPTIONS: unbound variable
```

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, and CI is not reporting issues. LGTM!

**Pull request merged**